app/screens/emoji_picker/picker/emoji_category_bar/index.tsx

```diff
@@ -45,22 +94,81 @@ const EmojiCategoryBar = ({onSelect}: Props) => {
-                    icon={icon.icon}
-                    index={index}
-                    scrollToIndex={scrollToIndex}
-                    theme={theme}
-                />
-            ))}
-        </View>
+        <>
+            {isEmojiPicker &&
+                <View
+                    style={styles.container}
+                    testID='emoji_picker.category_bar'
+                >
+                    {handleToggleEmojiPicker && deleteCharFromCurrentCursorPosition &&
+                    <View style={styles.categoryBar}>
+                        <View style={styles.keyboardControls}>
+                            <Pressable
+                                onPress={() => handleToggleEmojiPicker()}
+                                style={styles.actionIcons}
+                            >
+                                <CompassIcon
+                                    name={'keyboard-outline'}
+                                    size={20}
+                                    color={changeOpacity(theme.centerChannelColor, 0.56)}
+                                />
+                            </Pressable>
+                            <Divider
+                                orientation='vertical'
+                                color={changeOpacity(theme.centerChannelColor, 0.08)}
+                                width={1}
+                            />
+                        </View>
+                        <View style={{...styles.category, flex: isTablet ? 0 : 1}}>
+                            {!isTablet &&
+                            <ScrollView
+                                horizontal={true}
+                                showsHorizontalScrollIndicator={false}
+                                contentContainerStyle={styles.scrollView}
+                            >
+                                {iconCatergories}
+                            </ScrollView>
+                            }
+                            {isTablet && iconCatergories}
+                        </View>
+                        <View style={styles.keyboardControls}>
+                            <Divider
+                                orientation='vertical'
+                                color={changeOpacity(theme.centerChannelColor, 0.08)}
+                                width={1}
+                            />
+                            <Pressable
+                                onPress={() => deleteCharFromCurrentCursorPosition()}
+                                style={styles.actionIcons}
+                            >
+                                <CompassIcon
+                                    name={'backspace-outline'}
+                                    size={20}
+                                    color={changeOpacity(theme.centerChannelColor, 0.56)}
+                                />
+                            </Pressable>
+                        </View>
+                    </View>}
+                </View>}
```

Comment:
Does it make sense to move this to a different component, for the sake of readability here?